src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CodeFixVerifierHelper.cs

```diff
@@ -87,18 +87,27 @@ private static void VerifyMessageHelpLinkUri(DiagnosticAnalyzer analyzer)
             }
         }
 
-        public static (SourceText? analyzerConfig, IEnumerable<KeyValuePair<OptionKey, object?>> options) ConvertOptionsToAnalyzerConfig(string defaultFileExtension, OptionsCollection options)
+        public static (SourceText? analyzerConfig, IEnumerable<KeyValuePair<OptionKey, object?>> options) ConvertOptionsToAnalyzerConfig(string defaultFileExtension, string? explicitEditorConfig, OptionsCollection options)
         {
             if (options.Count == 0)
             {
-                return (null, options);
+                var result = explicitEditorConfig is object ? SourceText.From(explicitEditorConfig, Encoding.UTF8) : null;
+                return (result, options);
             }
 
             var optionSet = new OptionSetWrapper(options.ToDictionary<KeyValuePair<OptionKey, object?>, OptionKey, object?>(option => option.Key, option => option.Value));
             var remainingOptions = new List<KeyValuePair<OptionKey, object?>>();
 
             var analyzerConfig = new StringBuilder();
-            analyzerConfig.AppendLine("root = true");
+            if (explicitEditorConfig is object)
+            {
+                analyzerConfig.AppendLine(explicitEditorConfig);
+            }
+            else
+            {
+                analyzerConfig.AppendLine("root = true");
+            }
```

Comment:
Nit: I know we have differing styles, but i find this much more understandable as:

```suggestion
                analyzerConfig.AppendLine(explicitEditorConfig ?? "root = true");
```

Reply:
I don't see the two branches as doing the same thing, so I kept them separate.

Reply:
i personally find that strange.  they're appending the first line. a nd that first line is eithe hte editorconfig if we have it, or the `root = ...` line otherwise...   Structurally, semantically, and (Afaict) intentionally, they're the same.  the only difference is what line we're trying to append.  and that seems easily expressed with `x ?? y`.  `x` if we have it `y` otherwise... :)